presto-thrift-connector-api/src/main/java/com/facebook/presto/connector/thrift/api/datatypes/PrestoThriftBigint.java

```diff
@@ -0,0 +1,127 @@
+import com.facebook.presto.spi.type.Type;
+import com.facebook.swift.codec.ThriftConstructor;
+import com.facebook.swift.codec.ThriftField;
+import com.facebook.swift.codec.ThriftStruct;
+
+import javax.annotation.Nullable;
+
+import java.util.Arrays;
+import java.util.Objects;
+
+import static com.facebook.presto.connector.thrift.api.PrestoThriftBlock.bigintData;
+import static com.facebook.presto.spi.type.BigintType.BIGINT;
+import static com.facebook.swift.codec.ThriftField.Requiredness.OPTIONAL;
+import static com.google.common.base.MoreObjects.toStringHelper;
+import static com.google.common.base.Preconditions.checkArgument;
+
+/**
+ * Elements of {@code nulls} array determine if a value for a corresponding row is null.
+ * Elements of {@code longs} array are values for each row. If row is null then value is ignored.
+ */
+@ThriftStruct
+public final class PrestoThriftBigint
+        implements PrestoThriftColumnData
+{
+    private final boolean[] nulls;
+    private final long[] longs;
+
+    @ThriftConstructor
+    public PrestoThriftBigint(@Nullable boolean[] nulls, @Nullable long[] longs)
+    {
+        checkArgument(sameSizeIfPresent(nulls, longs), "nulls and values must be of the same size");
+        this.nulls = nulls;
+        this.longs = longs;
+    }
+
+    @Nullable
+    @ThriftField(value = 1, requiredness = OPTIONAL)
+    public boolean[] getNulls()
+    {
+        return nulls;
+    }
+
+    @Nullable
+    @ThriftField(value = 2, requiredness = OPTIONAL)
+    public long[] getLongs()
+    {
+        return longs;
+    }
+
+    @Override
+    public Block toBlock(Type desiredType)
+    {
+        checkArgument(BIGINT.equals(desiredType), "type doesn't match: %s", desiredType);
+        int numberOfRecords = numberOfRecords();
+        return new LongArrayBlock(
+                numberOfRecords,
+                nulls == null ? new boolean[numberOfRecords] : nulls,
+                longs == null ? new long[numberOfRecords] : longs);
+    }
+
+    @Override
+    public int numberOfRecords()
+    {
+        return nulls != null ? nulls.length : (longs != null ? longs.length : 0);
+    }
+
+    @Override
+    public boolean equals(Object obj)
+    {
+        if (this == obj) {
+            return true;
+        }
+        if (obj == null || getClass() != obj.getClass()) {
+            return false;
+        }
+        PrestoThriftBigint other = (PrestoThriftBigint) obj;
+        return Arrays.equals(this.nulls, other.nulls) &&
+                Arrays.equals(this.longs, other.longs);
+    }
+
+    @Override
+    public int hashCode()
+    {
+        return Objects.hash(Arrays.hashCode(nulls), Arrays.hashCode(longs));
+    }
+
+    @Override
+    public String toString()
+    {
+        return toStringHelper(this)
+                .add("numberOfRecords", numberOfRecords())
+                .toString();
+    }
+
+    public static PrestoThriftBlock fromSingleValueBlock(Block block)
+    {
+        if (block.isNull(0)) {
```

Comment:
`checkArgument(block.getPositionCount() == 1, "block does not have a single value")`